Library/Homebrew/cmd/update.sh

```diff
@@ -640,10 +631,21 @@ EOS
 
       # HOMEBREW_UPDATE_FORCE and HOMEBREW_UPDATE_AUTO aren't modified here so ignore subshell warning.
       # shellcheck disable=SC2030
-      if [[ "${UPSTREAM_REPOSITORY_URL}" == "https://github.com/"* ]]
+      if [[ "${UPSTREAM_REPOSITORY_URL}" =~ https://(([^:@]+)(:([^@]+))?@)?github.com/(.*)$ ]]
       then
-        UPSTREAM_REPOSITORY="${UPSTREAM_REPOSITORY_URL#https://github.com/}"
-        UPSTREAM_REPOSITORY="${UPSTREAM_REPOSITORY%.git}"
+        UPSTREAM_REPOSITORY="${BASH_REMATCH[5]%.git}"
+        # HOMEBREW_GITHUB_API_TOKEN is optionally defined in the user environment.
+        # can be superseded a token from the local repository URL
+        # shellcheck disable=SC2153
+        if [[ -n "${BASH_REMATCH[4]}" ]]
+        then
+          CURL_GITHUB_API_ARGS=("--header" "Authorization: token ${BASH_REMATCH[4]}")
+        elif [[ -n "${HOMEBREW_GITHUB_API_TOKEN}" ]]
+        then
+          CURL_GITHUB_API_ARGS=("--header" "Authorization: token ${HOMEBREW_GITHUB_API_TOKEN}")
+        else
+          CURL_GITHUB_API_ARGS=()
+        fi
```

Comment:
```suggestion
        # HOMEBREW_GITHUB_API_TOKEN is optionally defined in the user environment.
        # can be superseded a token from the local repository URL
        # shellcheck disable=SC2153
        if [[ "${UPSTREAM_REPOSITORY_URL}" == "https://github.com/"* ]] && 
           [[ -n "${HOMEBREW_GITHUB_API_TOKEN}" ]]
        then
          CURL_GITHUB_API_ARGS=("--header" "Authorization: token ${HOMEBREW_GITHUB_API_TOKEN}")
        elif [[ -n "${BASH_REMATCH[4]}" ]]
        then
          CURL_GITHUB_API_ARGS=("--header" "Authorization: token ${BASH_REMATCH[4]}")
        else
          CURL_GITHUB_API_ARGS=()
        fi
```

Reply:
@sazriel26 This change also doesn't seem to have been made. It's important that your new logic is a fallback to the existing logic, not a replacement. I do not trust it to behave consistently and any issues with this file mean people cannot get updates from Homebrew.